packages/perseus-score/src/empty-diner-check.test.ts

```diff
@@ -0,0 +1,288 @@
+                    ],
+                    buttonSets: [],
+                    functions: [],
+                    times: false,
+                },
+            },
+        },
+        images: {},
+    };
+}
+
+function generateBasicRadioQuestion(): PerseusRenderer {
+    return {
+        content: "[[☃ radio 1]]",
+        widgets: {
+            "radio 1": {
+                type: "radio",
+                options: {
+                    choices: [
+                        {
+                            content: "Correct",
+                            correct: true,
+                        },
+                        {
+                            content: "Incorrect",
+                            correct: false,
+                        },
+                    ],
+                },
+            },
+        },
+        images: {},
+    };
+}
+
+describe(`emptyDinerCheck`, () => {
+    it("passes an answered dropdown", () => {
+        const question = generateBasicDropdownQuestion();
+
+        const userInput: UserInputMap = {
+            "dropdown 1": {
+                value: 1,
+            },
+        };
+
+        expect(emptyDinerCheck(generatePerseusItem(question), userInput)).toBe(
+            false,
+        );
+    });
+
+    it("fails an unanswered dropdown", () => {
+        const question = generateBasicDropdownQuestion();
+
+        const userInput: UserInputMap = {
+            "dropdown 1": {
+                value: 0,
+            },
+        };
+
+        expect(emptyDinerCheck(generatePerseusItem(question), userInput)).toBe(
+            true,
+        );
+    });
+
+    it("passes any interactive graph", () => {
+        // No point in making a whole bunch of data
+        // IG doesn't do client-side validation
+        const question: any = {
+            question: {
+                content: "[[☃ interactive-graph 1]]",
+                widgets: {
+                    "interactive-graph 1": {},
+                },
+            },
+        };
+
+        const userInput: any = {};
+
+        expect(emptyDinerCheck(question, userInput)).toBe(false);
+    });
+
+    it("passes an answered numeric-input", () => {
+        const question = generateBasicNumericInputQuestion();
+
+        const userInput: UserInputMap = {
+            "numeric-input 1": {
+                currentValue: "42",
+            },
+        };
+
+        expect(emptyDinerCheck(generatePerseusItem(question), userInput)).toBe(
+            false,
+        );
+    });
+
+    it("fails an unanswered numeric-input", () => {
+        const question = generateBasicNumericInputQuestion();
+
+        const userInput: UserInputMap = {
+            "numeric-input 1": {
+                currentValue: "",
+            },
+        };
+
+        expect(emptyDinerCheck(generatePerseusItem(question), userInput)).toBe(
+            true,
+        );
+    });
+
+    it("passes an unanswered numeric-input (coefficient edge case)", () => {
```

Comment:
This is protective, but we don't expect these to show up for projects that exclusively use the DINER widgets, right?